rsconnect/tests/test_environment.py

```diff
@@ -32,55 +35,46 @@ def test_get_default_locale(self):
     def test_file(self):
         result = detect_environment(get_dir("pip1"))
 
-        pip_version = result.pop("pip")
-        self.assertTrue(version_re.match(pip_version))
+        self.assertTrue(version_re.match(result.pip))
 
-        locale = result.pop("locale")
-        self.assertIsInstance(locale, str)
-        self.assertIn(".", locale)
+        self.assertIsInstance(result.locale, str)
+        self.assertIn(".", result.locale)
 
-        self.assertEqual(
-            result,
-            {
-                "package_manager": "pip",
-                "source": "file",
-                "filename": "requirements.txt",
-                "contents": "numpy\npandas\nmatplotlib\n",
-                "python": self.python_version(),
-            },
-        )
+        for key, value in {
+            "package_manager": "pip",
+            "source": "file",
+            "filename": "requirements.txt",
+            "contents": "numpy\npandas\nmatplotlib\n",
+            "python": self.python_version(),
+        }.items():
+            self.assertEqual(getattr(result, key), value)
```

Comment:
nit: you could construct an expected Environment here and compare to the one you got back.

Reply:
ok, maybe not, forgot we pop things out before comparing

Reply:
no I think you're right! changing it...